### DISCUSSION TITLE
Espanso only partially working on openSUSE Tumbleweed

### DISCUSSION THREAD

**Discussion by johanneskastl** [espanso/espanso#2058] [Q&A]
Hi all,

I packaged Espanso for openSUSE and tested the package.

The good news is: It is working!

The bad news: I seem to only get it working in GVim, xterm and Signal. All other applications that I tried did not work.

I might be missing a dependency for them to work properly? Maybe some Plasma-related tool?

Currently xclip and xdotool are being installed as dependency.

If anyone has ideas where to start digging, any help is highly appreciated!

Kind Regards,
Johannes

## Working

- GVim
- xterm
- Signal Desktop (electron-based package, not a Flatpak)
- 

## Non-working applications

I tried both `:espanso` and `#detect#`, but espanso does not work in the following applications:

- Kate (KDE Plasma text editor)
- Konsole (the KDE Plasma terminal)
- Firefox
- Thunderbird
- LibreOffice Writer

**Comment by johanneskastl** [comment]
OK, I just found that Wayland support is experimental. No idea why I missed that part of the docs...

I'll dig through it and report back.

**Comment by smeech** [comment]
If it's working in some programs but not in others, that may be down to the [injection mechanism](https://espanso.org/docs/matches/basics/#injection-mechanism). Otherwise, try some of the relevant [options](https://espanso.org/docs/configuration/options/#options-reference), although the Wayland ones are rather limited.
